test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/AbstractLocalClusterFactory.java

```diff
@@ -216,6 +216,14 @@ public String getTransportEndpoint() {
             return readPortsFile(portsFile).get(0);
         }
 
+        public List<String> getAvailableTransportEndpoints() {
+            Path portsFile = workingDir.resolve("logs").resolve("transport.ports");
+            if (Files.notExists(portsFile)) {
+                return List.of();
+            }
```

Comment:
For my education: Why are we ok with this being empty, i.e. nodes are not started as said in comment on the interface? If the tests assert cluster formation, I assume all nodes should start? Or is there a timing issue?

Reply:
It's a timing issue indeed.